homeassistant/components/mqtt/__init__.py

```diff
@@ -260,41 +248,43 @@ async def _async_config_entry_updated(hass: HomeAssistant, entry: ConfigEntry) -
 
     Causes for this is config entry options changing.
     """
-    mqtt_client = hass.data[DATA_MQTT]
+    mqtt_data: MqttData = hass.data[DATA_MQTT]
```

Comment:
I can't review everything right now, but I think it would make sense to use a wrapper, so that you don't have to type it everywhere else. Something like...
```python
    def get_mqtt_data(hass: HomeAssistant) -> MqttData
        return hass.data[DATA_MQTT]
```

Reply:
I leave it as it is